packages/commonwealth/client/scripts/views/components/component_kit/new_designs/CWForm.tsx

```diff
@@ -14,49 +19,61 @@ type FormProps = {
   onWatch?: (values: any) => void;
 };
 
+export type CWFormRef = UseFormReturn;
+
 /**
  * Provides a wrapper around the HTML <form/> components. This wrapper
  * adds form validation to the nested fields with a simple to use API.
  */
-const CWForm = ({
-  id,
-  onSubmit = () => null,
-  onErrors = () => null,
-  validationSchema,
-  children,
-  className,
-  initialValues,
-  onWatch,
-}: FormProps) => {
-  const formMethods: UseFormReturn = useForm({
-    resolver: zodResolver(validationSchema),
-    defaultValues: initialValues,
-    mode: 'all',
-  });
-
-  useEffect(() => {
-    if (onWatch) {
-      const subscription = formMethods.watch(onWatch);
-      return () => subscription.unsubscribe();
-    }
-  }, [formMethods, onWatch]);
-
-  const handleFormSubmit = async (event) => {
-    // This will chain our custom onSubmit along with the react-hook-form's submit chain
-    await formMethods.handleSubmit(onSubmit)(event);
-
-    // trigger error callback if there are any errors
-    Object.keys(formMethods.formState.errors).length &&
-      (await onErrors(formMethods.formState.errors));
-  };
-
-  return (
-    <FormProvider {...formMethods}>
-      <form id={id} onSubmit={handleFormSubmit} className={className}>
-        {typeof children === 'function' ? children(formMethods) : children}
-      </form>
-    </FormProvider>
-  );
-};
+const CWForm = forwardRef<UseFormReturn, FormProps>(
+  (
+    {
+      id,
+      onSubmit = () => null,
+      onErrors = () => null,
+      validationSchema,
+      children,
+      className,
+      initialValues,
+      onWatch,
+    }: FormProps,
+    ref,
+  ) => {
+    const formMethods: UseFormReturn = useForm({
+      resolver: zodResolver(validationSchema),
+      defaultValues: initialValues,
+      mode: 'all',
+    });
+
+    // Expose formMethods to parent components using ref
+    useImperativeHandle(ref, () => formMethods);
```

Comment:
Marcin, first use of this hook in the app 😄

Reply:
goat!